### PR TITLE
Replace squirrel fork with `Clowd.Squirrel`

### PR DESCRIPTION
**Pull request by peppy**
Better maintained and much nicer structure. Also promising outlook in terms of features we need. Worth reading through their [readme](https://github.com/clowd/Clowd.Squirrel) to get an overview of what has changed.

This hooks in to https://github.com/ppy/osu/issues/15731. I see it as one step forward, but am still not 100% sure it will have all the features and platform support we require. That said, the developer is active and seemingly willing to work through issues. If we can get the functionality we need from this project it will take a large weight off my shoulders.

I've already gone through all the local hotfixes we had on our squirrel fork and they have been fixed, all in much cleaner ways.

PRing this to allow others to regression test. I've done initial testing but will be doing some more full release testing before unblocking.

### CODE REVIEW

**Comment by frenzibyte** [comment]
> I see it as one step forward, but am still not 100% sure it will have all the features and platform support we require.

Just to mention, I've went through the repository and it appears they do have an issue thread tracking adding support for macOS and Linux now that they're running on .NET 6 (https://github.com/clowd/Clowd.Squirrel/issues/17).

**Comment by peppy** [comment]
Yep. I've talked with the developer and it is a current focus (work has already been started) so that's promising.

File: osu.Desktop/Program.cs [hunk 1]
```diff
@@ -104,6 +110,25 @@ public static void Main(string[] args)
...
+            }, onAppUninstall: (version, tools) =>
+            {
+                tools.RemoveShortcutForThisExe();
+                tools.RemoveUninstallerRegistryEntry();
+            }, onEveryRun: (version, tools, firstRun) =>
+            {
+                tools.SetProcessAppUserModelId();
+
+                if (firstRun)
+                    tools.CreateUninstallerRegistryEntry();
```
Comment (bdach) [inline osu.Desktop/Program.cs:128]: Is there a particular reason why this is done on first run rather than on initial install?

**Review by bdach — COMMENTED** [review]
Going to need to test this on windows, just read the diff for now. Appears generally sane.

File: osu.Desktop/Program.cs (hunk 1 above)
Comment (peppy) [reply osu.Desktop/Program.cs:128]: I have no idea why this is here. I must have been sleep-coding.

**Comment by peppy** [comment]
I've done some further testing and am good with this.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**